jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -1008,6 +1016,10 @@ SSLConnectionSocketFactory getTwoWaySSLSocketFactory() throws SQLException {
         JdbcConnectionParams.SUNJSSE_ALGORITHM_STRING);
       String keyStorePath = sessConfMap.get(JdbcConnectionParams.SSL_KEY_STORE);
       String keyStorePassword = sessConfMap.get(JdbcConnectionParams.SSL_KEY_STORE_PASSWORD);
+      if (keyStorePassword == null) {
+        keyStorePassword = Utils.getPasswordFromCredentialProvider(
```

Comment:
Seems like we load all credentials from the same creds file, is this by design?

And we can add some comment how to use the `key` to generate the local creds, looks like we use `JdbcConnectionParams.SSL_KEY_STORE_PASSWORD` as the key.

Reply:
Can we move all these three types of passwords to a different API? This API can take in the `String key` as a parameter and return a string password. You can then call the API at lines #900, #1018, and #1037.

Reply:
> Seems like we load all credentials from the same creds file, is this by design?
> 
> And we can add some comment how to use the `key` to generate the local creds, looks like we use `JdbcConnectionParams.SSL_KEY_STORE_PASSWORD` as the key.

Yes. We store all the passwords in the same file. Will document about the generation and usage in cwiki https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients

Reply:
> Can we move all these three types of passwords to a different API? This API can take in the `String key` as a parameter and return a string password. You can then call the API at lines #900, #1018, and #1037.

Done. Added an API in Utils.java that takes map and key as arguments. First tries to find the password from map. If not found, then gets it from credential provider.